src/Aspire.Hosting/IDistributedApplicationBuilder.cs

```diff
@@ -16,6 +16,11 @@ public interface IDistributedApplicationBuilder
     /// <inheritdoc cref="HostApplicationBuilder.Configuration" />
     public ConfigurationManager Configuration { get; }
 
+    /// <summary>
+    /// Directory of the project where the app host is located. Defaults to the content root if there's no project.
+    /// </summary>
+    public string ProjectDirectory { get; }
```

Comment:
@mitchdenny this cleans up *ALOT* and it's what your docker file PR should be using.